pkg/container/device.go

```diff
@@ -163,3 +164,41 @@ func (c *container) specAddContainerConfigDevices() error {
 	}
 	return nil
 }
+
+// getDeviceUID() and getDeviceGID() are used to find the right
+// uid/gid values for the device node created in the container
+// namespace. The runtime executes mknod() and chmod()s the created
+// device with the values returned here.
+//
+// TODO(mythi): In case of user namespaces, the runtime simply bind
+// mounts the devices from the host. Additional logic is needed
+// to check that the runtimes effective UID/GID on the host has the
+// permissions to access the device node and/or the right user namespace
+// mappings are created.
+//
+// CRI-O has an experimental support for setting user namespace mappings
+// via annotations when pod's securitycontext runs as root/uid=0. When
+// enabled, the logic below does not change the behavior for existing
+// mappings unless container's securitycontext user/group overrides what
+// is set for the pod.
+//
+// Ref: https://github.com/kubernetes/kubernetes/issues/92211
+func getDeviceUID(config *types.ContainerConfig, hostUID uint32, enableDeviceOwnershipFromSecurityContext bool) *uint32 {
+	if userval := config.Linux.SecurityContext.RunAsUser; userval != nil {
+		uid := uint32(userval.Value)
+		if uid > 0 && enableDeviceOwnershipFromSecurityContext {
```

Comment:
can we refactor this to use a helper function for the common bits? we can pass a UInt64Value field into the helper:
```
func getID(userval UInt64Val, enableDeviceOwnershipFromSecurityContext bool) {
  ...
}
```

Reply:
fixed!